### PR TITLE
[Engine][PhysicalPlan] Recreate action with parallelism

### PR DESCRIPTION
**Pull request by Hisoka-X**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/execution/ExecutionPlanGenerator.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ private void createExecutionVertex(LogicalVertex logicalVertex) {
         final long newId = idGenerator.getNextId();
         Action newAction;
         if (chainedVertices.size() < 1) {
-            newAction = recreateAction(logicalVertex.getAction(), newId);
+            newAction = recreateAction(logicalVertex.getAction(), newId, logicalVertex.getParallelism());
         } else {
             List<SeaTunnelTransform> transforms = new ArrayList<>(chainedVertices.size());
             List<String> names = new ArrayList<>(chainedVertices.size());
```
Comment (ashulin) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/execution/ExecutionPlanGenerator.java:163]: Need to change line 178?

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/execution/ExecutionPlanGenerator.java:163]: No, this bug only happen when use union feature.

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/execution/ExecutionPlanGenerator.java:163]: No, this bug only happen when use union feature.

**Review by ashulin — APPROVED** [review]
LGTM

**Pull request merged**